math/mathmore/src/complex_quartic.h

```diff
@@ -150,21 +153,14 @@ gsl_poly_complex_solve_quartic (double a, double b, double c, double d,
 
         disc = R2 - Q3;
```

Comment:
I am not sure having a fix tolerance is always correct. If R2 and Q3 are both very small, smaller than ZeroTolerance, their difference will be significantly different than zero.

Reply:
Do you think a relative check for `disc` would be better? On the other hand, if the numbers are small (in the order `1e-16` small), I think it's very likely to run into issues in other places... In any case, I'd really like to get this fixed before the holiday, right now our CI has a failing test!

Reply:
What about adding something like this, which protects the case the numbers are large. Probably the case of smaller value is excluded for numerical precision in the computations of R2 and Q3 which are results from a subtraction
```
static constexpr double ZeroTolerance = 8 * std::numeric_limits<double>::epsilon(); 
auto norm = std::min((std::abs(R2) + std::abs(Q3)), std::numeric_limits<double>::max());
if (disc < std::max( std::numeric_limits<double>::epsilon(), ZeroTolerance * norm) {.......
  //   R2 and Q3 are considered equal here....
```

Reply:
I implemented a subset of your suggestion as `ZeroTolerance * (fabs(R2) + fabs(Q3))`. The `std::min(/* ... */, std::numeric_limits<double>::max())` part doesn't make sense to me since it will always just evaluate to the first argument. Regarding `epsilon()`, see above.

Reply:
I agree the first check, `std::min(std::abs(R2) + std::abs(Q3)), std::numeric_limits<double>::max()` is maybe not really needed, but it does not harm, it is a protection in case R2 and Q3 are very large. 
But the second check (std::max) I think is needed. It is a switch between absolute and relative tolerance.

Reply:
It still doesn't make sense to me: The maximum can, by definition, never be exceeded, so I don't see how this ever applies.